src/tmx/TmxUtils/src/telemetry/PluginTelemetry.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+#include <string>
+#include <iostream>
+#include <tmx/TmxException.hpp>
+#include <tmx/messages/message.hpp>
+using namespace std;
+
+namespace tmx::utils::telemetry
+{
+    struct PluginInfo
+    {
+        string id;
+        //Brief name for a plugin
+        string name; 
+        //Detailed description of the plugin and its usage
+        string description;
+        //Latest version of the plugin
+        string version;
+    };
+
+    struct PluginInstallation
+    {  
+        //Whether the plugin is current enabled or disabled. When a plugin is not installed, enabled default value is set to -1
+        int enabled = -1;
+        //Where the plugin executable is located
+        string path;
+        //The plugin executable name
+        string exeName;
+        //The manifest file that contains some configuration parameters
+        string manifest;
+        //Message broadcast interval
+        string maxMessageInterval;
+        string commandLineParameters;
+    };
+
+    class PluginTelemetry
+    {
+    private:
+        PluginInfo _pluginInfo;
+        PluginInstallation _pluginInstallation;
+    public:
+        PluginTelemetry()=default;
+        PluginInfo getPluginInfo() const;
+        PluginInstallation getPluginInstallation() const;
+        void setPluginInfo(const PluginInfo& pluginInfo);
+        void setPluginInstallation(const PluginInstallation& pluginInstallation);
+         /***
+         * @brief Serialize PluginTelemetry object into JSON container (tmx::message_container_type)
+         * @return A JSON container of tmx::message_container_type
+         */
+        tmx::message_container_type serialize() const;
```

Comment:
I think the naming for this method is also misleading. Serialize implies we are taking a data structure and converting it into a format for sending or storing like json or bytes. In this case we are really only converting this data type to and from property tree. I think we could name this in that fashion `toPtree` or `fromPtree`. Another option would be to use naming convention similar to what other data structures in v2xhub use that use property trees. I think they use methods `set_contents` and `get_contents` to set the object based on a property tree or get the property tree of the data in a object,